### ISSUE TITLE
dash::Shared with Atomic Min / Max Operations Leads to Incorrect Results

### ISSUE THREAD

**Issue by rkowalewski** [dash-project/dash#544]
The following code does not work, at least with extreme (edge case) values. Sometimes it works, sometimes not. I tried to find the root cause on the quick, however, failed. A pull request with the corresponding unit test will be available in couple of minutes.

~~~
auto&    team = dash::Team::All();
shared_t g_min{dash::team_unit_t{0}, team};
shared_t g_max(dash::team_unit_t{0}, team);

g_min.get().fetch_op(dash::min<int>(), 0);
g_max.get().fetch_op(dash::max<int>(), std::numeric_limits<int>::max());

g_min.barrier();
g_max.barrier();

auto const min = static_cast<int>(g_min.get());
auto const max = static_cast<int>(g_max.get());
~~~

**Comment by rkowalewski** [comment]
I suspect there is something strange happening in DART / MPI since the DASH side looks right at a first glance.

If the issue is fixed we should [revert this commit](https://github.com/dash-project/dash/pull/543/commits/b1501f7c00be104156982641e863947fe54442a8) and replace two-sided min/max with one-sided min/max reduction.

**Comment by rkowalewski** [comment]
fixed by #545
